src/brsTypes/components/RoSGNode.ts

```diff
@@ -344,6 +364,51 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
             return child;
         },
     });
+
+    /* Returns true if the subject node has the remote control focus, and false otherwise */
+    private hasfocus = new Callable("hasfocus", {
+        signature: {
+            args: [],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter) => {
+            return BrsBoolean.from(interpreter.environment.getFocusedNode() === this);
+        },
+    });
+
+    /**
+     *  If on is set to true, sets the current remote control focus to the subject node,
+     *  also automatically removing it from the node on which it was previously set.
+     *  If on is set to false, removes focus from the subject node if it had it
+     */
+    private setfocus = new Callable("setfocus", {
+        signature: {
+            args: [new StdlibArgument("on", ValueKind.Boolean)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, on: BrsBoolean) => {
+            interpreter.environment.setFocusedNode(on.toBoolean() ? this : BrsInvalid.Instance);
+            return BrsBoolean.False; //brightscript always returns false for some reason
+        },
+    });
+    /**
+     *  Returns true if the subject node or any of its descendants in the SceneGraph node tree
+     *  has remote control focus
+     */
+    private isinfocuschain = new Callable("isinfocuschain", {
```

Comment:
`isInFocusChain` will return `false` if this nodes parent has focus? I believe you just was never sure on what the `focus chain` meant

Reply:
yeah that's what I'd expect.  isInFocusChain() is anything current node and its children, where as hasFocus() is just the current node